src/compiler/tsc.ts

```diff
@@ -415,7 +415,8 @@ namespace ts {
 
             const compileResult = compile(rootFileNames, compilerOptions, compilerHost);
 
-            if (!compilerOptions.watch) {
+            // Firefox has Object.prototype.watch
+            if (!compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")) {

```

Comment:
I think the problem is that that's still an API breaking change, and adds non-uniformity to our command line parser.